### DISCUSSION TITLE
Support for Plantower PTQS1005

### DISCUSSION THREAD

**Discussion by wapebira** [arendst/Tasmota#14593] [Ideas]
PTQS1005, like PMSx003 uses UART for communication. It requires RX to be connected to ESP TX as does not support passive mode as other PMSX003. A command (0xAC) needs to be sent for receiving the data. 
Here is the data sheet: [http://www.sodocs.net/doc/f85c6d86940590c69ec3d5bbfd0a79563c1ed4fc-8.html](url)
Here is an python library implementation: [https://git.quacker.org/d/pyptqs1005/src/branch/master/libptqs1005/__init__.py](url)
Is it possible to be implemented, or there is a workaround I am not aware?

**Comment by barbudor** [comment]
If you plan to use an ESP32, it should be easy to port the Python driver to the Berry language
Alternatively it seems possible to extend the existing PMS5003/PMS3003 driver to support this model

**Comment by wapebira** [comment]
Yes. Will be used with ESP32 or ESP8266.

**Comment by kindomLee** [comment]
Any progress on intergrate ptqs1005 with Tasmota?

**Reply by sfromis** [reply]
Roadmaps for future additions is not how it works. Most new device drivers are created when someone with a need decides to do the work, and submit a PR for inclusion in the project.
